cedar-lean/Cedar/Thm/Termination/Expr.lean

```diff
@@ -0,0 +1,118 @@
+/-
+ Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Spec.Expr
+
+namespace Cedar.Spec
+
+open Cedar.Data
+
+def Expr.size : Expr -> Nat
```

Comment:
You shouldn't need to define this `size` function.  Lean automatically generates these for you, so you can write `sizeOf e` when `e` is an Expr.